cores/arduino/WString.cpp

```diff
@@ -492,7 +492,13 @@ int String::compareTo(const String &s) const
 		if (buffer && len > 0) return *(unsigned char *)buffer;
 		return 0;
 	}
-	return strcmp(buffer, s.buffer);
+	const char *p1 = buffer;
+	const char *p2 = s.buffer;
+
+	while (*p1 == *p2++)
+		if ('\0' == *p1++)
+			return 0;
+	return (*(const char *)p1 - *(const char *)(p2 - 1));
```

Comment:
Agree with Calvin.  It may require more work.  However, for future referencing and categorizing, it is desirable to separate out issues into individual reports.  It will also ensure the discussion, analysis of the issues will not be mixed together.